src/tree/fit_stump.cc

```diff
@@ -43,8 +44,12 @@ void FitStump(Context const* ctx, linalg::TensorView<GradientPair const, 2> gpai
     }
   }
   CHECK(h_sum.CContiguous());
-  collective::Allreduce<collective::Operation::kSum>(
-      reinterpret_cast<double*>(h_sum.Values().data()), h_sum.Size() * 2);
+
+  // In vertical federated learning, only worker 0 needs to call this, no need to do an allreduce.
```

Comment:
Maybe we run it for all workers to reduce a condition? We have a `validate` method in learner model param, which is a good place for checking whether the label is correctly distributed across workers for federated learning. If labels are the same for all workers, the base_score should also be the same. Also, we don't need an additional info parameter.

Reply:
The issue is in `learner.cc` we only call `InitEstimation` for worker 0, which in turn calls this method. If we don't skip this allreduce we'd get a mismatch in non-0 workers.